lib/jsdom/living/traversal/TreeWalker-impl.js

```diff
@@ -0,0 +1,197 @@
+
+			while (!isNull(sibling)) {
+				node = sibling;
+				const result = this._filterNode(node);
+				if (result === FILTER_ACCEPT) return (this._currentNode = node);
+				sibling = (type === NEXT ? node.firstChild : node.lastChild);
+
+				if (result === FILTER_REJECT || isNull(sibling)) {
+					sibling = (type === NEXT ? node.nextSibling : node.previousSibling);
+				}
+			}
+
+			node = node.parentNode;
+			if (isNull(node) || node === this.root) return null;
+			if (this._filterNode(node) === FILTER_ACCEPT) return null;
+		}
+	};
+
+	constructor(args, privateData) {
+		this.root = privateData.root;
+		this.whatToShow = privateData.whatToShow;
+		this.filter = privateData.filter;
+		//this.expandEntityReferences = privateData.expandEntityReferences;  //removed from living
+		this.currentNode = this.root;
+	};
+
+	get currentNode() {
+		return this._currentNode;
+	};
+
+	set currentNode(node) {
+		if (isNull(node)) throw new DOMException(DOMException.NOT_SUPPORTED_ERR, "Cannot set currentNode to null");
+		this._currentNode = node;
+	};
+
+	parentNode() {
+		let node = this._currentNode;
+		while (!isNull(node) && node !== this.root) {
+			node = node.parentNode;
+			if (!isNull(node) && this._filterNode(node) === FILTER_ACCEPT) return (this._currentNode = node);
+		}
+		return null;
+	};
+
+	firstChild() {
+		return this._traverseChildren(FIRST);
+	};
+
+	lastChild() {
+		return this._traverseChildren(LAST);
+	};
+
+	previousSibling() {
+		return this._traverseSiblings(PREVIOUS);
+	};
+
+	nextSibling() {
+		return this._traverseSiblings(NEXT);
+	};
+
+	previousNode() {
+		let node = this._currentNode;
+
+		while (node !== this.root) {
+			let sibling = node.previousSibling;
+
+			while (!isNull(sibling)) {
+				node = sibling;
+				let result = this._filterNode(node);
+
+				while (result !== FILTER_REJECT && node.hasChildNodes()) {
+					node = node.lastChild;
+					result = this._filterNode(node);
+				}
+
+				if (result === FILTER_ACCEPT) return (this._currentNode = node);
+				sibling = node.previousSibling;
+			}
+
+			if (node === this.root || isNull(node.parentNode)) return null;
+			node = node.parentNode;
+			if (this._filterNode(node) === FILTER_ACCEPT) return (this._currentNode = node);
+		}
+
+		return null;
+	};
+
+	nextNode() {
+		let node = this._currentNode;
+		let result = FILTER_ACCEPT;
+
+		while (true) {
+			while (result !== FILTER_REJECT && node.hasChildNodes()) {
+				node = node.firstChild;
+				result = this._filterNode(node);
+				if (result === FILTER_ACCEPT) return (this._currentNode = node);
+			}
+
+			//if (isNull((node = idlUtils.wrapperForImpl(domSymbolTree.following(node, {root: this.root}))))) return null;
+			do {
+				if (node === this.root) return null;
+				const sibling = node.nextSibling;
+
+				if (!isNull(sibling)) {
+					node = sibling;
+					break;
+				}
+
+				node = node.parentNode;
+			} while (!isNull(node));
+
+			if (isNull(node)) return null;
```

Comment:
On a side note, you should mention that contributors should run `npm run lint` before sending a PR in your [Contributing.md](https://github.com/tmpvar/jsdom/blob/master/Contributing.md) document.